asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9.8-extension-timestamp.adoc

```diff
@@ -0,0 +1,86 @@
+
+* epoch versioning,
+* optional epoch time-step offsets,
+* optional versioning of `pm:CalibrationInfo/@Time`, `pm:AlertSystemState/@LastSelfCheck`, `pm:AlertConditionState/@DeterminationTime`, `pm:AbstractMetricValue/@StartTime`, `pm:AbstractMetricValue/@StopTime`, `pm:AbstractMetricValue/@DeterminationTime`, `pm:AbstractContextState/@BindingStartTime` and `pm:AbstractContextState/@BindingEndTime`. 
+
+[sdpi_level=+1]
+====== Model
+
+The clock epoch schema is available in <<vol3_appendix_a_xml_schemas_timestamp_version>>. <<vol3_example_extension_clock_discontinuities>> shows an exemplary XML instance of a <<vol2_clause_dev_30_message_getmdibresponse, {var_label_dev_30_message_getmdibresponse}>> from a device that has experienced two recent non-slewing time adjustments following three adjustments some time in the past. Of particular note:
+
+* the clock state includes epoch time-step offsets for epochs 3 and 4; earlier versions are not referenced and therefore not required,
+* the state for metric `m1` references epoch version 3; all timestamps in this state are versioned,
+* the timestamp for metric `m2` is not versioned; its timestamp is less than `pm:ClockState/@LastSet` and its value should be treated with greater suspicion than later timestamps, 
+* although the current time (`pm:ClockState/@DateAndTime`) is also less than `pm:ClockState/@LastSet`, the current time is always reported using the current time-reference frame; its value need not be treated with any more suspicion than normal,
+* each `sdpi:Epoch` includes a `@Version`, `@Timestamp` and `@Offset`; the timestamp is in the time-reference frame of the versioned epoch, adding the `@Timestamp` and `@Offset` provides a timestamp for an equivalent point in time for the next epoch version,
+* the default value of any timestamp not specifically versioned is the current epoch version. 
+
+.Example MDIB state following two recent non-slewing time adjustments
+[#vol3_example_extension_clock_discontinuities]
+====
+[source,xml]
+----
+include::../../listings/vol3-clause-biceps-content-example-timestamp-version.xml[]
+----
+====
+
+[sdpi_level=+1]
+====== Requirements
+
+.R0600
+[sdpi_requirement#r0600,sdpi_req_level=shall]
+****
+The <<vol1_spec_sdpi_p_actor_somds_provider>> shall increment `sdpi:Epochs/@Version` by exactly one, beginning from 0, for every non-slewing time adjustment to any system clock used as part of its System Function Contribution (<<acronym_sfc>>). 
+
+****
+
+.R0601
+[sdpi_requirement#r0601,sdpi_req_level=shall]
+****
+A <<vol1_spec_sdpi_p_actor_somds_provider>> that versions timestamps in any `pm:AbstractMetricValue`, `pm:AbstractContextState`, `pm:AlertSystemState`, `pm:CalibrationInfo` and/or `pm:AlertConditionState` shall include, in every clock state update, the complete history of epoch offsets from the earliest version referenced in the MDIB to the current time-reference frame version.  
```

Comment:
I suppose this should be scoped to MDIB sequences

Reply:
Renumbered and added as a new requirement modelled after R0047 from 10207.